app/src/main/java/tech/relaycorp/gateway/data/model/ParcelDirection.kt

```diff
@@ -0,0 +1,11 @@
+package tech.relaycorp.gateway.data.model
+
+enum class ParcelDirection(val value: String) {
+    Incoming("incoming"), Outgoing("outgoing");
```

Comment:
I'd suggest to avoid the terms "incoming" and "outgoing" as that's ambiguous in a broker like this gateway, where its responsibility is to route data.

This is the terminology I used on the public gateway to avoid ambiguity:

- "(Private) gateway-bound parcels": Parcels bound for a private gateway. Those parcels will be encapsulated in cargo if a courier picks them up, or sent as is when using PoWebSockets.
- "Internet-bound parcels": Parcels bound for an endpoint on the Internet. Behind the scenes, that endpoint could be a service (e.g., `https://api.twitter.com/relaynet`) or another gateway (e.g., `https://public-gateway.relaycorp-competitor.com`) if the recipient of the parcel is a private endpoint.

However, I don't think they'd work here because things will be more complex in private gateways:

- One optimisation we need to do after this project is to deliver parcels bound for public Internet endpoints directly, skipping the public gateway, when the Internet is available and the public endpoint is not blocked by the ISP. We'll only send the parcel to our public gateway when we can't deliver it directly.
- We're also going to support other underlying networks in the medium term, not just the Internet and courier-based sneakernets. For example, we'll eventually support Bluetooth-based meshnets, where parcels could be broadcast to all gateways on the meshnet or sent to a specific gateway -- and those parcels will never be sent to the Internet or a courier.

For the reasons above, I'd suggest the following terms for now (but I'm very much open to suggestions):

- "Local endpoint-bound parcels".
- "Internet endpoint-bound parcels".

Thoughts?

Reply:
... In other words, I don't think we should be thinking about the _direction_ of the parcel, but I think we should be thinking about the _location of its target endpoint_:

- Local network.
- Internet.
- Meshnet (outside current scope).
- Etc.

Reply:
Ok, definitely more confusing to think of all those scenarios right now 😅

So a kind of `recipientLocation` or `targetLocation`?

Reply:
I think so! I'd prefer the former because it has the word "recipient" in it.